spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/Neo4jHealthIndicator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.health;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.neo4j.ogm.model.Result;
+import org.neo4j.ogm.session.Session;
+import org.neo4j.ogm.session.SessionFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.Assert;
+
+/**
+ * {@link HealthIndicator} that tests the status of a Neo4j by executing a Cypher
+ * statement.
+ *
+ * @author Eric Spiegelberg
+ */
+public class Neo4jHealthIndicator extends AbstractHealthIndicator {
+
+	/** The key used to store Cypher execution results within the Health.Builder. */
+	public static final String CYPHER_RESULTS = "cypherResults";
+
+	/** The default Cypher query. */
+	public static final String CYPHER_DEFUALT = "match (n) return count(n) as nodeCount";
+
+	private SessionFactory sessionFactory;
+
+	private Map<String, Object> emptyParameters = new HashMap<>();
+
+	/**
+	 * The default Cypher statement.
+	 */
+	@Value("${neo4jHealthIndicator.cypher}")
```

Comment:
I don't know what this is but we don't use `@Value`. Check the codebase for similar use cases (there are other health indicators that expose configuration keys).